bentoml/_internal/bento/build_config.py

```diff
@@ -132,6 +129,12 @@ def _convert_env(
 
 @attr.frozen
 class DockerOptions:
+
+    # User shouldn't add new fields under yaml file.
+    __forbid_extra_keys__ = True
+    # no need to omit since BentoML has already handled the default values.
```

Comment:
```suggestion
    # always omit config values in case of default values got changed in future BentoML releases
```
```suggestion
    # no need to omit since BentoML has already handled the default values.
```
```suggestion
    # no need to omit since BentoML has already handled the default values.
```